### DISCUSSION TITLE
WalletConnect Provider - eth_sendTransaction does not exist

### DISCUSSION THREAD

**Discussion by 0xSandwich** [WalletConnect/walletconnect-monorepo#1780] [Developer Support]
Hello,

I'm working on a dapp using web3modal v2 with wagmi. 
Everything works fine for read functions such as getting maxSupply, mintPrice etc..
However for transaction calls it doesn't work, I get an error 
`eth_sendTransaction does not exist/is not available`

I initialize walletconnect provider as shown in the docs on my app root file:
```
  // Wagmi client
  const { provider } = configureChains(chains, [
    walletConnectProvider({ projectId: config.walletConnectProjectId }),
  ]);
  const wagmiClient = createClient({
    autoConnect: true,
    connectors: modalConnectors({ appName: "web3Modal", chains }),
    provider,
  });
```

Then on my custom hook I create the contract instance using that same provider:
`
    const provider = useProvider({chainId:5})

    // Contract instance
    useEffect(() => {
        if (provider && address) {
            const importPromise = import(`../artifacts/contracts/${config.contract.name}.sol/${config.contract.abiName}.json`)
            importPromise.then((contract) => {
                const contractInst = new ethers.Contract(config.contract.address, contract.abi, provider.getSigner(address))
                setContract(contractInst)
            })
        }
    },[provider,address])
`

How to make transactions functions working with web3modal v2 ?
On an older project using walletconnect, I was set RPC provider and it was working. Can't find the way to do it with web3modal v2
```
const provider = new WalletConnectProvider({
    rpc: {
        43114: "https://api.avax.network/ext/bc/C/rpc"
    },
    chainId:43114,
});  
```

**Comment by xzilja** [comment]
I think issue could be caused by `walletConnectProvider` (which is our rpc), not supporting avax. So you basically need create new JSONRpc provider for avalanche (https://wagmi.sh/react/providers/jsonRpc) and add it to your wagmi client config.

Also, you can probably simplify your logic for contracts by also using wagmi there i.e for contract writes https://wagmi.sh/react/hooks/useContractWrite
